lib/date_time/date_time.c

```diff
@@ -263,8 +265,15 @@ static int current_time_check(void)
 		return -ENODATA;
 	}
 
+	/* Allow elapsed time from last time update to be slightly shorter than configured value.
+	 * The reason is that otherwise the update would happen at every second update interval
+	 * because last_date_time_update is taken after the update was done and this comparison
+	 * occurs before time update. Hence there is a gap of some milliseconds.
+	 */
 	if ((k_uptime_get() - time_aux.last_date_time_update) >
-	    CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS * 1000) {
+	    CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS * MSEC_PER_SEC -
```

Comment:
This should be casted to (int) or otherwise it's being treated as unsigned causing this branch never to be hit with 0 interval

Reply:
Done